Hide/Toggle left sidebar

zhongyi-tong opened the pull request:
![c](https://user-images.githubusercontent.com/7262715/28461047-871e22de-6dda-11e7-9efc-e7c9d7db964b.gif)

Fix #222.

Commit:
Add collapse button.

Commit:
Add CollapseButton component.

Commit:
Reduce action button's tooltip size.

akashnimare commented:
Looks good. Can we set a better position of arrow? because we don't want to overlap app's content with arrow-

![image](https://user-images.githubusercontent.com/2263909/28461619-12b8fc6a-6e35-11e7-8ec9-6926e79f3404.png)
![image](https://user-images.githubusercontent.com/2263909/28461633-1f547936-6e35-11e7-8b76-2e8e2971e025.png)

zhongyi-tong commented:
But where... I couldn't think of a better place. What about move it to the top? But it covers the content anyway...

akashnimare commented:
@brockwhittaker, @timabbott can you suggest a better position?

timabbott commented:
Well, when the left sidebar is visible, we could easily put a widget in the upper-left corner, inside the box containing the Zulip logo.  In smaller windows, it's a lot harder; that's probably a rare case for the desktop app but I could imagine having it cover over the top half of the hamburger menu?

I think this is a good topic to post on #design for ideas.  

One thing to keep in mind is that we could integrate this more tightly with the web code and offer support for more directly controlling what happens.

akashnimare commented:
Yeah, we can easily put it in the upper-left corner when the **width is > 776 px.** The problem comes when the window size is smaller than this.

Commit:
Add settings item for sidebar toggle.

Commit:
Revert collapse button.

Commit:
Initialize sidebar appearance from settings.

zhongyi-tong commented:
New Design (with a config item and a menu item):
![image](https://user-images.githubusercontent.com/7262715/28616190-6f5b0cb2-71c1-11e7-9a84-39da7eb0df5e.png)

Commit:
Add menu item for sidebar toggle.

akashnimare commented on app/renderer/js/pages/preference/general-section.js:
```diff
@@ -120,6 +132,22 @@ class GeneralSection extends BaseComponent {
+		this.$sidebarOptionSettings = document.querySelector('#ui-option-settings #sidebar-option .setting-control');
+		this.$sidebarOptionSettings.innerHTML = '';
+
+		const sidebarOption = ConfigUtil.getConfigItem('show-sidebar', false);
```
It would be better if we show sidebar initially. If we hide it on the first load, users might not able to find it. What do you think?

akashnimare replied:
@geeeeeeeeek something is wrong. No matter what I set here it's getting `null` initially.

akashnimare committed:
Show sidebar initially

We should not hide it by default since users might not able to find it. Let's show it initially and let them decide.

akashnimare commented:
Merged. Thanks, @geeeeeeeeek.
